### PR TITLE
Use network-bsd unconditionally and support GHC 9.0

### PR DESCRIPTION
**Pull request by chshersh**

### CODE REVIEW

File: Database/MongoDB/Admin.hs [hunk 1]
```diff
@@ -76,7 +76,7 @@ renameCollection from to = do
     db <- thisDatabase
     useDb admin $ runCommand ["renameCollection" =: db <.> from, "to" =: db <.> to, "dropTarget" =: True]
 
-dropCollection :: (MonadIO m) => Collection -> Action m Bool
+dropCollection :: (MonadIO m, MonadFail m) => Collection -> Action m Bool
```
Comment (TheOddler) [inline Database/MongoDB/Admin.hs:79]: Why is this extra constraint needed without any other code-changes here?

**Review by TheOddler — APPROVED** [review]
LGTM :D

File: Database/MongoDB/Admin.hs (hunk 1 above)
Comment (arianvp) [reply Database/MongoDB/Admin.hs:79]: in newer version of GHC the `Monad` typeclass doesn't have a `fail :: String -> m ()` function anymore. It instead was split off into a new typeclass `MonadFail`

**Comment by chshersh** [comment]
Closing this PR because we should use `mongoDB` from Stackage that already has what we need. Ideally, it would be nice to archive this repository and delete eventually when we're confident enough we don't use it anymore.

**Pull request closed without merging**